### PR TITLE
Less logging and event handling.

### PR DESCRIPTION
**Pull request by amrsoll**
One more stab at reducing log traffic.

### CODE REVIEW

File: octoprint_mrbeam/gcodegenerator/jobtimeestimation.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def on_event(self, event, payload):
             estimation_thread.daemon = True
             estimation_thread.start()
 
-        if event == OctoPrintEvents.CLIENT_OPENED:
+        if event == OctoPrintEvents.CLIENT_OPENED and self._last_estimation != -1:
```
Comment (andywerner) [inline octoprint_mrbeam/gcodegenerator/jobtimeestimation.py:70]: What problem does this solve or what is the benefit?

File: octoprint_mrbeam/iobeam/iobeam_handler.py [hunk 2]
```diff
@@ -485,13 +485,14 @@ def _work(self):
...
-                                self._logger.warn(
-                                    "ANDYTEST sock_data: %s", len(sock_data)
+                            if (
+                                len(temp_buffer) >= self.MESSAGE_LENGTH_MAX
+                                or len(sock_data) >= self.MESSAGE_LENGTH_MAX - 5
+                            ):
+                                self._logger.info(
+                                    "Receiving long message - buffer size: %s, receiving size : %s",
+                                    len(temp_buffer),
+                                    len(sock_data),
```
Comment (andywerner) [inline octoprint_mrbeam/iobeam/iobeam_handler.py:495]: I think we can remove this completely. Let's quickly discuss a potential socket saturation/overflow.

File: octoprint_mrbeam/iobeam/iobeam_handler.py [hunk 3]
```diff
@@ -512,22 +513,15 @@ def _work(self):
...
-                            # Split all JSON messages by new line character
-                            messages = data.split(self.MESSAGE_NEWLINE)
-
-                            if not data.endswith(self.MESSAGE_NEWLINE):
-                                # Record remaining part of data into temp buffer, to read messages longer than MESSAGE_LENGTH_MAX
-                                temp_buffer = messages.pop()
-                            else:
-                                temp_buffer = b""
+                        # Split all JSON messages by new line character
+                        messages = data.split(self.MESSAGE_NEWLINE)
```
Comment (andywerner) [inline octoprint_mrbeam/iobeam/iobeam_handler.py:517]: what exactly is the difference here?

**Review by andywerner — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/gcodegenerator/jobtimeestimation.py (hunk 1 above)
Comment (amrsoll) [reply octoprint_mrbeam/gcodegenerator/jobtimeestimation.py:70]: Fewer events. This one is entirely unnecessary if no estimations had been made before-hand.

If an estimation is being calculated, then we want to notify front end, but let's check if there is one running first.

File: octoprint_mrbeam/iobeam/iobeam_handler.py (hunk 2 above)
Comment (amrsoll) [reply octoprint_mrbeam/iobeam/iobeam_handler.py:495]: I agree.

File: octoprint_mrbeam/iobeam/iobeam_handler.py (hunk 3 above)
Comment (amrsoll) [reply octoprint_mrbeam/iobeam/iobeam_handler.py:517]: You already made a check if data existed or not, so I removed the redundant check.

File: octoprint_mrbeam/gcodegenerator/jobtimeestimation.py (hunk 1 above)
Comment (amrsoll) [reply octoprint_mrbeam/gcodegenerator/jobtimeestimation.py:70]: TODO : Check if a "ready to laser state" is not communicated properly when reloading the page

File: octoprint_mrbeam/iobeam/iobeam_handler.py (hunk 2 above)
Comment (andywerner) [reply octoprint_mrbeam/iobeam/iobeam_handler.py:495]: pls remove the logging

File: octoprint_mrbeam/gcodegenerator/jobtimeestimation.py (hunk 1 above)
Comment (andywerner) [reply octoprint_mrbeam/gcodegenerator/jobtimeestimation.py:70]: ok, fine with me

Comment (amrsoll) [reply octoprint_mrbeam/gcodegenerator/jobtimeestimation.py:70]: Develop branch did not display the ready to laser page when reloading prior to the merge. Therefore it is not blocking.

**Pull request merged**